main/src/library/Chain.flix

```diff
@@ -0,0 +1,865 @@
+
+    ///
+    /// Helper function for `indexOf`.
+    ///
+    def indexOfHelper(a: a, acc: Int32, vx: ViewLeft[a]): Int32 with Eq[a] = match vx {
+        case NoneLeft        => -1
+        case SomeLeft(x, xs) => if (x == a) acc else indexOfHelper(a, acc + 1, viewLeft(xs))
+    }
+
+    ///
+    /// Alias for `findLeft`.
+    ///
+    /// The function `f` must be pure.
+    ///
+    pub def find(f: a -> Bool, xs: Chain[a]): Option[a] = findLeft(f, xs)
+
+    ///
+    /// Optionally returns the first element of `xs` that satisfies the predicate `f` when searching from left to right.
+    ///
+    /// The function `f` must be pure.
+    ///
+    pub def findLeft(f: a -> Bool, xs: Chain[a]): Option[a] = match viewLeft(xs) {
+        case NoneLeft        => None
+        case SomeLeft(x, rs) => if (f(x)) Some(x) else findLeft(f, rs)
+    }
+
+    ///
+    /// Optionally returns the first element of `xs` that satisfies the predicate `f` when searching from right to left.
+    ///
+    /// The function `f` must be pure.
+    ///
+    pub def findRight(f: a -> Bool, xs: Chain[a]): Option[a] = match viewRight(xs) {
+        case NoneRight        => None
+        case SomeRight(rs, x) => if (f(x)) Some(x) else findRight(f, rs)
+    }
+
+    ///
+    /// Returns a list of all integers between `b` (inclusive) and `e` (exclusive).
+    ///
+    /// Returns `Nil` if `b >= e`.
+    ///
+    pub def range(b: Int32, e: Int32): Chain[Int32] = rangeHelper(b, e, Chain.empty())
+
+    ///
+    /// Helper function for `range`.
+    ///
+    def rangeHelper(b: Int32, e: Int32, acc: Chain[Int32]): Chain[Int32] =
+        if (b >= e) acc else rangeHelper(b + 1, e, snoc(acc, b))
+
+    ///
+    /// Returns a list with the element `a` repeated `n` times.
+    ///
+    /// Returns `Nil` if `n < 0`.
+    ///
+    pub def repeat(n: Int32, a: a): Chain[a] = repeatHelper(n, a, Empty)
+
+    ///
+    /// Helper function for `repeat`.
+    ///
+    def repeatHelper(n: Int32, a: a, acc: Chain[a]): Chain[a] =
+        if (n <= 0)
+            acc
+        else {
+            let ac1 = cons(a, acc);
+            repeatHelper(n-1, a, ac1)
+        }
+
+    ///
+    /// Alias for `scanLeft`.
+    ///
+    pub def scan(f: (b, a) -> b & ef, s: b, xs: Chain[a]): Chain[b] & ef = scanLeft(f, s, xs)
+
+    ///
+    /// Accumulates the result of applying `f` to `xs` going left to right.
+    ///
+    /// That is, the result is of the form: `s :: f(s, x1) :: f(f(s, x1), x2)  ...`.
+    ///
+    pub def scanLeft(f: (b, a) -> b & ef, s: b, xs: Chain[a]): Chain[b] & ef = scanLeftHelper(f, s, xs, singleton(s))
+
+    ///
+    /// Helper function for `scanLeft`.
+    ///
+    def scanLeftHelper(f: (b, a) -> b & ef, s: b, xs: Chain[a], acc: Chain[b]): Chain[b] & ef = match viewLeft(xs) {
+        case NoneLeft        => acc
+        case SomeLeft(x, rs) => {let a = f(s, x); scanLeftHelper(f, a, rs, snoc(acc, a))}
```

Comment:
I notice that you sometimes write it like the above, and other times you break it into multiple lines. 

You don't have to change it, just wanted to remark it :)

Reply:
I don't have a preference for either (otherwise they would be consistent).

In general it's nice if cases can be one liners, but obviously that isn't always possible.